### PR TITLE
WIP: Make this repository a ROS package

### PR DESCRIPTION
**Pull request by VictorLamoine**
# :construction: Work in progress :building_construction: 

## Travis
[Build Status](https://travis-ci.org/InstitutMaupertuis/robodk_postprocessors)

## ROS package
- [ ] Code an action server
- [ ] Add tests for this server

## Misc
- [ ] Enable Travis CI builds for this repository
- [ ] Change Travis CI image badge

### CODE REVIEW

**Comment by VictorLamoine** [comment]
I don't have the privileges to enable Travis for this repository. If you have please enable Travis builds for this repository!

https://travis-ci.org/ros-industrial

**Comment by gavanderhoorn** [comment]
Something to consider: should this be a ROS package, or should we maintain it as a Python library that can then be used by a ROS package?

The former doesn't really add that much (apart from being able to 'build' it (whatever that means for Python) with Catkin), while the latter keeps things ROS-agnostic and cross-platform.

Testing can still be done using something like `nosetest`, and there are plenty of resources on how to configure CI (ie: Travis) for regular Python packages.

**Comment by VictorLamoine** [comment]
It's hosted under `ROS-Industrial` so I'm in favor of making this repository a ROS package.
We could make it easy to use it without ROS if we keep your comment in mind.

If we decide that this repository must not depend on ROS then we need to create an other one and I'm afraid it might cause confusion as to what is this repository.

# Possible solutions
- We could add a simple script to remove everything ROS related and transform this repository into a simple python library.
- Maintain a "master" branch that contains a ROS-agnostic python library (then we need to backport improvements etc.)

**Comment by gavanderhoorn** [comment]
@VictorLamoine wrote:
> It's hosted under `ROS-Industrial` so I'm in favor of making this repository a ROS package.
We could make it easy to use it without ROS if we keep your comment in mind.

I don't really see why everything under 'ros-industrial' should necessarily be a ros pkg. We have other repositories that are not packages as well.

> If we decide that this repository must not depend on ROS then we need to create an other one and I'm afraid it might cause confusion as to what is this repository.

This repository could basically be a stand-alone Python library, which is then used as a *system dependency* by an actual ROS package. I realise I'm not very representative of potential target users, but I don't think that would be much more complicated than how we deal with other system dependencies right now?

**Comment by VictorLamoine** [comment]
Ok let's do it this way I agree there are some advantages in keeping the ROS package and the python scripts separated.

I'll keep this PR open and work inside it until I have something working. We'll advice later on a new repository name etc.

The ROS package will include a submodule containing this repository, and that's fine!

**Comment by shaun-edwards** [comment]
Perhaps @nubi86 could chime in.  Remember we want to keep things in a usable form for RoboDK.  Otherwise I don't see how this is valuable for them.

**Comment by gavanderhoorn** [comment]
Good point.

**Comment by nubi86** [comment]
On my side it would be much better to keep each post processor as a
standalone module.
Could we have some sort of a wrapper that links ROS with the standalone
post processors?

Otherwise, if you believe this is of no use in its current form feel free
to go ahead and change what you think is appropriate.
We can also adjust future versions of RoboDK to any
modifications/improvements, or re-writing some methods to make it more
intuitive (if they can somehow remain Standalone).

**Comment by AustinDeric** [comment]
Excellent work.  We should 100% keep it as a python library and make a ROS wrapper, i think that general approach is best for most things.  However, if y'all want to make is a ROS package, then thats fine too.

I would like to start using @VictorLamoine's code immediatly, so can we merge it in to the kinetic-devel branch and make all outstanding items into issues?

move fast and break things.

Thanks!

File: scripts/generate_program_server.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env python
+
+import rospy
+
+import actionlib
```
Comment (AustinDeric) [inline scripts/generate_program_server.py]: This should be a very short execution.  Why not implement a service?

File: README.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ## RoboDK Post Processors
+
+[![Build Status](https://travis-ci.org/InstitutMaupertuis/robodk_postprocessors.svg?branch=kinetic-devel)](https://travis-ci.org/InstitutMaupertuis/robodk_postprocessors)
```
Comment (AustinDeric) [inline README.md]: change to ros-industrial travis-ci.

File: package.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  <url type="repository">https://github.com/ros-industrial/robodk_postprocessors</url>
+  <url type="bugtracker">https://github.com/ros-industrial/robodk_postprocessors/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <depend>actionlib</depend>
+  <depend>actionlib_msgs</depend>
+  <depend>genmsg</depend>
+  <depend>roscpp</depend>
+  <depend>message_generation</depend>
```
Comment (AustinDeric) [inline package.xml]: build_depend

File: package.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  <depend>message_runtime</depend>
```
Comment (AustinDeric) [inline package.xml]: exec_depend

File: package.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>robodk_postprocessors</name>
+  <version>0.1.0</version>
+  <description>RoboDK generic industrial post processors</description>
+  <author>Albert Nubiola</author>
+  <maintainer email="victor.lamoine@gmail.com">Victor Lamoine</maintainer>
+  <license>BSD</license>
```
Comment (AustinDeric) [inline package.xml]: Apache2.0

File: package.xml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+  <url type="repository">https://github.com/ros-industrial/robodk_postprocessors</url>
+  <url type="bugtracker">https://github.com/ros-industrial/robodk_postprocessors/issues</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
+  <depend>actionlib</depend>
+  <depend>actionlib_msgs</depend>
+  <depend>genmsg</depend>
+  <depend>roscpp</depend>
```
Comment (AustinDeric) [inline package.xml]: rospy

**Comment by VictorLamoine** [comment]
Closing as we want to keep this repository clean and instead do something like https://github.com/AustinDeric/ros_industrial_codegen

**Pull request closed without merging**